linker/shared/src/test/scala/org/scalajs/linker/testutils/LinkingUtils.scala

```diff
@@ -72,8 +72,9 @@ object LinkingUtils {
       implicit ec: ExecutionContext): Future[ModuleSet] = {
 
     val cfg = config.withCheckIR(true)
-    val frontend = StandardLinkerFrontend(cfg)
+
     val backend = new StoreModuleSetLinkerBackend(StandardLinkerBackend(cfg))
+    val frontend = StandardLinkerFrontend(cfg, backend.symbolRequirements)
```

Comment:
This is an example of the burden that this PR would shift on users that intercept parts of the linker pipeline (not terrible IMO, but it does raise the question if the current abstractions are right).